packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -574,10 +574,15 @@ export default createRule<Options, MessageId>({
         node.type === AST_NODE_TYPES.MemberExpression
           ? !isNullableOriginFromPrev(node)
           : true;
+      const unionParts = unionTypeParts(type);
+      const possiblyVoid = unionParts.some(
+        part => part.flags & ts.TypeFlags.Void,
+      );
       return (
         isTypeAnyType(type) ||
         isTypeUnknownType(type) ||
```

Comment:
Interesting: `isTypeAnyType(type) and `isTypeUnknownType(type)` both are working fine even if the type is a union. Both of those functions narrow down to checking `isTypeFlagSet(type, ts.TypeFlags.Something)`.

I tried switching from the `possiblyVoid` variable to checking `isTypeFlagSet(type, ts.TypeFlags.Void` - and it seems to work! So I think my hunch about using `unionTypeParts` was wrong 😄.

Tip: TypeScript's flags enums such as `ts.TypeFlags` work with checks like `isTypeFlagSet(type, ts.TypeFlags.One | ts.TypeFlags.Two)`. So you should be able to replace `isTypeAnyType(type) || isTypeUnknownType(type)` with a single `isTypeFlagSet` check.

Reply:
This is great to know! Done.